From 4932aa05595a99321ef77cb013a8f7b03fbef156 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Wed, 16 Apr 2014 10:02:32 -0700
Subject: [PATCH 1/2] cleanup: fix compiler warnings

Clean up some of our patches that caused compiler warnings.
---
 gdhcp/client.c     |    2 +-
 include/inet.h     |    1 +
 plugins/ethernet.c |    4 +++-
 src/connman.h      |    3 +++
 src/service.c      |   26 ++++++++++++++++++++++----
 5 files changed, 30 insertions(+), 6 deletions(-)

diff --git a/include/inet.h b/include/inet.h
index a836a5b..8a0c3a6 100644
--- a/include/inet.h
+++ b/include/inet.h
@@ -50,6 +50,7 @@ int connman_inet_del_host_route(int index, const char *host);
 int connman_inet_add_network_route(int index, const char *host, const char *gateway,
 					const char *netmask);
 int connman_inet_del_network_route(int index, const char *host);
+int connman_inet_del_network_route2(int index, const char *host, const char *gateway, const char *netmask);
 int connman_inet_clear_gateway_address(int index, const char *gateway);
 int connman_inet_set_gateway_interface(int index);
 int connman_inet_clear_gateway_interface(int index);
diff --git a/plugins/ethernet.c b/plugins/ethernet.c
index e70c2fc..d1ac209 100644
--- a/plugins/ethernet.c
+++ b/plugins/ethernet.c
@@ -27,6 +27,8 @@
 #include <net/if.h>
 #include <string.h>
 #include <stdio.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
 #include <linux/sockios.h>
 #include <linux/if_vlan.h>
 
@@ -93,7 +95,7 @@ static struct connman_network_driver eth_network_driver = {
 	.disconnect	= eth_network_disconnect,
 };
 
-int get_vlan_vid(const char *ifname)
+static inline int get_vlan_vid(const char *ifname)
 {
 	int vid;
 	struct vlan_ioctl_args ifr;
diff --git a/src/connman.h b/src/connman.h
index cc5b445..ec2a98c 100644
--- a/src/connman.h
+++ b/src/connman.h
@@ -773,6 +773,9 @@ void __connman_service_notify(struct connman_service *service,
 int __connman_service_counter_register(const char *counter);
 void __connman_service_counter_unregister(const char *counter);
 
+void __connman_service_set_routes(struct connman_service *service, char **routes);
+void add_routes_config(int index, struct connman_service *service);
+
 #include <connman/peer.h>
 
 int __connman_peer_init(void);
diff --git a/src/service.c b/src/service.c
index 889be46..49a79d7 100644
--- a/src/service.c
+++ b/src/service.c
@@ -31,6 +31,8 @@
 #include <ctype.h>
 #include <stdint.h>
 #include <stdlib.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
 
 #include <linux/if_vlan.h>
 #include <linux/sockios.h>
@@ -1935,6 +1937,22 @@ static void del_vlan_by_vid(struct connman_service *service, const int vid)
 		del_vlan(vlan_service);
 }
 
+static inline int get_vlan_vid(const char *ifname)
+{
+	int vid;
+	struct vlan_ioctl_args ifr;
+	memset(&ifr, '\0', sizeof(ifr));
+
+	int fd = socket(AF_INET, SOCK_STREAM, 0);
+	if (fd < 0)
+		return -errno;
+	ifr.cmd = GET_VLAN_VID_CMD;
+	strncpy(ifr.device1, ifname, sizeof(ifr.device1));
+	vid = ioctl(fd, SIOCSIFVLAN, &ifr) ? -errno : ifr.u.VID;
+	close(fd);
+	return vid;
+}
+
 /* Create NEW VLAN */
 static void add_vlan_by_vid(struct connman_service *service, const int vid)
 {
@@ -1968,14 +1986,14 @@ static void add_vlan_by_vid(struct connman_service *service, const int vid)
 static int update_vids_strings(DBusMessageIter *iter,
 				int *vids, char ***vids_strings)
 {
-	int i;
+	int i = 0;
 
 	if (*vids_strings) {
 		g_strfreev(*vids_strings);
 		*vids_strings = NULL;
 	}
 	if (vids) {
-		for (i = 0; vids[i] >= 0; i++) {
+		while (vids[i] >= 0) {
 			*vids_strings = g_renew(char *, *vids_strings, i + 2);
 			(*vids_strings)[i] = g_strdup_printf("%d", vids[i]);
 			(*vids_strings)[i + 1] = NULL;
@@ -1984,6 +2002,7 @@ static int update_vids_strings(DBusMessageIter *iter,
 				dbus_message_iter_append_basic(iter,
 						DBUS_TYPE_STRING,
 						&(*vids_strings)[i]);
+			i++;
 		}
 	}
 
@@ -3891,7 +3910,7 @@ static DBusMessage *set_property(DBusConnection *conn,
 	} else if (g_str_equal(name, "Routes.Configuration") == TRUE) {
 		DBusMessageIter entry;
 		GString *str;
-		int count = 0, index;
+		int index;
 
 		if (service->immutable == TRUE)
 			return __connman_error_not_supported(msg);
@@ -3909,7 +3928,6 @@ static DBusMessage *set_property(DBusConnection *conn,
 			const char *val;
 			char *colon, *slash, save;
 			int mask = 32;
-			int target;
 
 			dbus_message_iter_get_basic(&entry, &val);
 			dbus_message_iter_next(&entry);
-- 
1.7.10.4

